keps/sig-cloud-provider/20190422-cloud-controller-manager-migration.md

```diff
@@ -157,53 +159,35 @@ will prevent any of the v1.18 CCMs from claiming the lock. When the current hold
 
 ### Implementation Details/Notes/Constraints [optional]
 
-TODO after KEP summary and proposal is approved
+The cloud-provider needs to would need to implement an interface, that for a given controller returns the version of migration
 
 ### Risks and Mitigations
 
-TODO after KEP summary and proposal is approved
+One of the risks related to this KEP is the KCM and the CCM running the same controllers if the implementation is not well tested.
+To avoid this we should have e2e tests that covers the happy path and the edge cases
```

Comment:
Not against this but it seems like we are phrasing the problem we are trying to solve as one of the risks. If we are going to mention this I would put it in terms of the most likely circumstances. The one which occurs to me is that we have 5 different resources upon which the lock can be implemented, if the KCM and CCM do not agree then the locking mechanism will fail to work. (Eg KCM uses endpoints and CCM uses configmaps or leases).

I do think we should add a few other risks.
1) Volume of lease renew calls causing problems. We know that in most clusters leas renews account for over 10% of the traffic to the API Server.
2) Increases possibility of Controller Manager instability. Currently when a process loses its lease it kills itself. Currently we only hold a single lock. If we hold more locks then there is a potential for an increased rate of Controller Managers killing themselves.